capsules/src/alarm.rs

```diff
@@ -63,8 +65,13 @@ impl<A: Alarm<'a>> AlarmDriver<'a, A> {
             });
         }
         if next_alarm != u32::max_value() {
-            self.alarm.set_alarm(next_alarm);
-            Some(next_alarm)
+            if has_expired(next_alarm, now, set_at) {
+                self.fired();
```

Comment:
I think this has actually come up recently in https://github.com/tock/tock/issues/1496, where we cannot potentially issue a callback from a command or other function call. The `fired()` call has to occur in response to some interrupt. We can also use the deferred call mechanism to emulate an interrupt if need be.

Reply:
Brad is correct. The typical approach to this is to instead set minimum delay alarm. I.e., rather than invoke self.fired(), set next_alarm to be a very small value.